src/module/actor/character/sheet.ts

```diff
@@ -321,22 +321,32 @@ class CharacterSheetPF2e<TActor extends CharacterPF2e> extends CreatureSheetPF2e
                 (l) => l in CONFIG.PF2E.languages,
             );
             const isOverMax = languagesBuild.value > languagesBuild.max;
-            const languages: LanguageSheetData[] = actor.system.details.languages.value
-                .map((language) => {
-                    const label = game.i18n.localize(CONFIG.PF2E.languages[language] ?? language);
-                    const overLimit = isOverMax && sourceLanguages.indexOf(language) + 1 > languagesBuild.max;
-                    const tooltip = overLimit ? "PF2E.Actor.Character.Language.OverLimit" : null;
-                    return { slug: language, label, tooltip, overLimit };
-                })
-                .sort((a, b) => a.label.localeCompare(b.label));
+            const languageSlugs = actor.system.details.languages.value;
+            const commonLanguage = game.pf2e.settings.campaign.languages.commonLanguage;
+            const localizedLanguages: LanguageSheetData[] = languageSlugs.flatMap((language) => {
+                if (language === commonLanguage && languageSlugs.includes("common")) {
+                    return [];
+                }
+                const label =
+                    language === "common" && commonLanguage
+                        ? game.i18n.format("PF2E.Actor.Creature.Language.CommonLanguage", {
+                              language: game.i18n.localize(CONFIG.PF2E.languages[commonLanguage]),
+                          })
+                        : game.i18n.localize(CONFIG.PF2E.languages[language]);
+                const overLimit = isOverMax && sourceLanguages.indexOf(language) + 1 > languagesBuild.max;
```

Comment:
Does this handle dynamic positioning of the common language? Like if Taldane was the last language in the list, would it fail to mark one of the overlimit languages?